router/http/src/main/scala/io/buoyant/router/http/MethodAndHostIdentifier.scala

```diff
@@ -24,21 +24,23 @@ case class MethodAndHostIdentifier(
   private[this] def mkPath(path: Path): Dst.Path =
     Dst.Path(prefix ++ path, baseDtab(), Dtab.local)
 
-  def apply(req: http.Request): Future[(Dst, http.Request)] = req.version match {
-    case http.Version.Http10 =>
-      Future.value(
-        (mkPath(Path.Utf8("1.0", req.method.toString) ++ suffix(req)), req)
-      )
-
-    case http.Version.Http11 =>
-      req.host match {
-        case Some(host) if host.nonEmpty =>
+  def apply(req: http.Request): Future[(Dst, http.Request)] =
+    Identifier.dstConcrete(req, baseDtab).getOrElse {
```

Comment:
This seems a little surprising to me. If I specify a prefix on the identifier, I'd expect this prefix to apply to all names provided by the identifier.